pkg/api/v1/status/atlasstreaminstance.go

```diff
@@ -0,0 +1,56 @@
+package status
+
+import "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/common"
+
+type AtlasStreamInstanceStatus struct {
+	Common `json:",inline"`
+	// Unique 24-hexadecimal character string that identifies the instance
+	ID string `json:"id,omitempty"`
+	// List that contains the hostnames assigned to the stream instance.
+	Hostnames []string `json:"hostnames,omitempty"`
+	// List of connections configured in the stream instance.
+	Connections []StreamConnection `json:"connections,omitempty"`
+}
+
+type StreamConnection struct {
+	// Human-readable label that uniquely identifies the stream connection
+	Name string `json:"name,omitempty"`
+	// Reference for the resource that contains connection configuration
+	ResourceRef common.ResourceRefNamespaced `json:"resourceRef,omitempty"`
+}
+
+// +kubebuilder:object:generate=false
+
+type AtlasStreamInstanceStatusOption func(s *AtlasStreamInstanceStatus)
+
+func AtlasStreamInstanceDetails(ID string, hostnames []string) AtlasStreamInstanceStatusOption {
+	return func(s *AtlasStreamInstanceStatus) {
+		s.ID = ID
+		s.Hostnames = hostnames
+	}
+}
+
+func AtlasStreamInstanceAddConnection(name string, ref common.ResourceRefNamespaced) AtlasStreamInstanceStatusOption {
```

Comment:
I am worried that this will lead to status overload. Imagine we have thousands of stream instances pointing to a connection, this will imply thousands of status entries.

How about we just write a summary, i.e. "this connection is referenced by 1000 instances" ?

Reply:
This is the status of the instance, while it would also be possible to have 1000 connections referred by an instance, seems unmanageable from the perspective of the processor.
Anyway, the reason to have it here is that in the event of failure of Atlas API to sync a resource the Status should reflect what is actually in Atlas, i.e., if a connection is removed from spec and Atlas API fails to remove it, it will still be in the status, reflecting this connection is still part of the instance.

Reply:
oh i see, it's the reverse, nevermind then.